### PR TITLE
setup repo with latest push template code

### PR DESCRIPTION
**Pull request by rymorale**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PravinPK — APPROVED** [review]

**Review by spoorthipujariadobe — APPROVED** [review]

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AutoCarouselNotificationBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.AutoCarouselPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.CarouselPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a auto carousel push template notification.
+ */
+internal object AutoCarouselNotificationBuilder {
+    private const val SELF_TAG = "AutoCarouselTemplateNotificationBuilder"
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AutoCarouselNotificationBuilder.kt:34]: `AutoCarouselNotificationBuilder`

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/BasicNotificationBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.extensions.addActionButtons
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a basic push template notification.
+ */
+internal object BasicNotificationBuilder {
+    private const val SELF_TAG = "BasicTemplateNotificationBuilder"
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/BasicNotificationBuilder.kt:36]: `BasicNotificationBuilder`

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.CarouselPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.ManualCarouselPushTemplate
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.caching.CacheService
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a manual or filmstrip carousel push template notification.
+ */
+internal object ManualCarouselNotificationBuilder {
+    private const val SELF_TAG = "ManualCarouselTemplateNotificationBuilder"
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilder.kt:38]: `ManualCarouselNotificationBuilder`

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ZeroBezelNotificationBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import com.adobe.marketing.mobile.notificationbuilder.R
+import com.adobe.marketing.mobile.notificationbuilder.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.ZeroBezelPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+internal object ZeroBezelNotificationBuilder {
+    private const val SELF_TAG = "BasicTemplateNotificationBuilder"
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ZeroBezelNotificationBuilder.kt:29]: `ZeroBezelNotificationBuilder`

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/PushTemplateConstants.kt [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+package com.adobe.marketing.mobile.notificationbuilder.internal
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * This object holds all constant values for handling out-of-the-box push template notifications
+ */
+internal object PushTemplateConstants {
+    const val LOG_TAG = "AEPSDKPushTemplates"
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/PushTemplateConstants.kt:20]: `PushTemplateConstants`

File: code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/PushTemplateConstants.kt [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    internal object IntentActions {
+        const val FILMSTRIP_LEFT_CLICKED = "filmstrip_left"
+        const val FILMSTRIP_RIGHT_CLICKED = "filmstrip_right"
+        const val REMIND_LATER_CLICKED = "remind_clicked"
+        const val MANUAL_CAROUSEL_LEFT_CLICKED = "manual_left"
+        const val MANUAL_CAROUSEL_RIGHT_CLICKED = "manual_right"
+        const val INPUT_RECEIVED = "input_received"
+    }
+
+    internal object IntentKeys {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/PushTemplateConstants.kt:84]: TODO in next PR: let's make this object along with the enclosing one public

**Pull request merged**